nd4j/nd4j-backends/nd4j-tests/src/test/java/org/nd4j/linalg/factory/ops/NDBaseTest.java

```diff
@@ -0,0 +1,923 @@
+        //TODO: Fix. Not getting the expected output.
+        assertTrue(y.equalsWithEps(y_exp, 1e-6));
+    }
+
+    @Test
+    public void testFill() {
+        NDBase base = new NDBase();
+        INDArray x = Nd4j.createFromArray(2, 2);
+        INDArray y = base.fill(x, DataType.DOUBLE, 1.1);
+        INDArray y_exp = Nd4j.createFromArray(new double[][]{{1.1, 1.1}, {1.1, 1.1}});
+        // TODO: Fails: y is a float, not a double as expected.
+        assertTrue(y.equalsWithEps(y_exp, 1e-1));
+    }
+
+    @Test
+    public void testGather() {
+        NDBase base = new NDBase();
+        INDArray x = Nd4j.zeros(DataType.DOUBLE, 3, 3);
+        int[] ind = new int[]{0};
+        INDArray y = base.gather(x, ind, 0);
+        //TODO: crashes here. Op needs fixing.
+    }
+
+    @Test
+    public void testGatherNd() {
+        NDBase base = new NDBase();
+
+        assertTrue(false); // TODO: Fix the Op. (current signature is wrong.)
+    }
+
+    @Test
+    public void testScalarGt() {
+        //Scalar gt.
+        NDBase base = new NDBase();
+        INDArray x = Nd4j.zeros(DataType.DOUBLE, 3, 3);
+        INDArray y = base.gt(x, -0.1);
+        INDArray y_exp = Nd4j.createFromArray(new boolean[][]{{true, true, true}, {true, true, true}, {true, true, true}});
+        assertEquals(y, y_exp);
+        //TODO: Passes, but I do not like the detailed memory management in the Op constructor.
+    }
+
+    @Test
+    public void testGt() {
+        //element wise  gt.
+        NDBase base = new NDBase();
+        INDArray x = Nd4j.zeros(DataType.DOUBLE, 3, 3);
+        INDArray x1 = Nd4j.ones(DataType.DOUBLE, 3, 3);
+        INDArray y = base.gt(x1, x);
+        INDArray y_exp = Nd4j.createFromArray(new boolean[][]{{true, true, true}, {true, true, true}, {true, true, true}});
+        assertEquals(y, y_exp);
+    }
+
+
+    @Test
+    public void testScalarGte() {
+        //Scalar gte.
+        NDBase base = new NDBase();
+        INDArray x = Nd4j.zeros(DataType.DOUBLE, 3, 3);
+        INDArray y = base.gte(x, -0.1);
+        INDArray y_exp = Nd4j.createFromArray(new boolean[][]{{true, true, true}, {true, true, true}, {true, true, true}});
+        assertEquals(y, y_exp);
+        //TODO: Passes, but I do not like the detailed memory management in the Op constructor.
+    }
+
+    @Test
+    public void testGte() {
+        //element wise  gte.
+        NDBase base = new NDBase();
+        INDArray x = Nd4j.zeros(DataType.DOUBLE, 3, 3);
+        INDArray x1 = Nd4j.ones(DataType.DOUBLE, 3, 3);
+        INDArray y = base.gte(x1, x);
+        INDArray y_exp = Nd4j.createFromArray(new boolean[][]{{true, true, true}, {true, true, true}, {true, true, true}});
+        assertTrue(y.equals(y_exp));
+    }
+
+    @Test
+    public void testIdentity() {
+        NDBase base = new NDBase();
+        INDArray x = Nd4j.zeros(DataType.DOUBLE, 3, 3);
+        INDArray y = base.identity(x);
+        assertTrue(y.equalsWithEps(x, 1e-1));
```

Comment:
For stuff that should be exact (like identity) don't use equalwithEps, just use assertEquals(x, y)